src/macaron/slsa_analyzer/git_url.py

```diff
@@ -191,67 +175,16 @@ def check_out_repo_target(git_obj: Git, branch_name: str = "", digest: str = "",
     return True
 
 
-def commit_exists(git_obj: Git, digest: str) -> bool:
-    """Return True if the commit exists in the history of the current branch.
-
-    Parameters
-    ----------
-    git_obj : Git
-        The pydriller.Git wrapper object of the target repository.
-    digest : str
-        The hash of the commit that we want to check.
-
-    Returns
-    -------
-    bool
-    """
-    try:
-        return bool(git_obj.repo.is_ancestor(digest, "HEAD"))
-    except GitCommandError as error:
-        # The exception could be raised because the digest does not exist in the history of the branch
-        # or the digest is not a valid digest.
-        logger.debug(error)
-        return False
-
-
-def pull_latest_changes(git_obj: Git) -> bool:
-    """Pull the latest changes from remote.
-
-    This will perform git pull --ff-only on the current active branch of ``git_obj``. This will
-    leave HEAD pointing to the latest commit of the current branch.
-
-    If the repo is a shallow clone, this method will unshallow it before pulling the
-    latest changes.
-
-    Parameters
-    ----------
-    git_obj : Git
-        The pydriller.Git wrapper object of the target repository.
-
-    Returns
-    -------
-    bool
-        True if successful, else False.
-    """
-    try:
-        # Handle the case when the repository is a shallow cloned.
-        # https://git-scm.com/docs/git-rev-parse#Documentation/git-rev-parse.txt---is-shallow-repository
-        if str(git_obj.repo.git.rev_parse("--is-shallow-repository")).strip() == "true":
-            logger.info("The repository is a shallow clone. Trying to unshallow it.")
-            git_obj.repo.git.fetch("--unshallow")
-
-        # Pull the latest changes on the current branch fast-forward only.
-        git_obj.repo.git.pull("-f", "--ff-only")
-        return True
-    except GitCommandError as error:
-        logger.error(error)
-        return False
-
-
 def get_default_branch(git_obj: Git) -> str:
-    """Return the default branch of the target repository.
-
-    This function does not perform any online operation.
+    """Return the default branch name of the target repository.
+
+    This function does not perform any online operation. It depends on the existence of the remote
+    reference ``origin/HEAD`` in the git repository. This remote reference will point to the default
+    branch of the remote repository and it's usually set when the repository is first cloned with
+    ``git clone <url>``.
+    Therefore, this method will fail to obtain the default branch name if ``origin/HEAD`` is not
+    available. An example of this case is when a repository is shallow-cloned with a non-default the branch
```

Comment:
```suggestion
    available. An example of this case is when a repository is shallow-cloned from a non-default branch
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/491/commits/7b35e849e44cc4bdfd4b23499db6c16430b55270